cli/azd/internal/appdetect/java.go

```diff
@@ -2,21 +2,17 @@ package appdetect
 
 import (
 	"context"
-	"encoding/xml"
-	"fmt"
 	"io/fs"
 	"log"
-	"os"
 	"path/filepath"
-	"regexp"
 	"strings"
 
 	"github.com/azure/azure-dev/cli/azd/internal/tracing"
 	"github.com/azure/azure-dev/cli/azd/internal/tracing/fields"
 )
 
 type javaDetector struct {
-	rootProjects      []mavenProject
+	parentPoms        []pom
```

Comment:
I plan to delete parentPoms in next PR. The information in parentPoms will be merged into pom. And the pom will work like `effective pom`.